src/Microsoft.AspNetCore.ResponseCaching/KeyProvider.cs

```diff
@@ -153,5 +151,28 @@ public virtual string CreateVaryKey(HttpContext httpContext, VaryRules varyRules
                 _builderPool.Return(builder);
             }
         }
+
+        public virtual string CreateBodyKey(HttpContext httpContext)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            var builder = _builderPool.Get();
+
+            try
+            {
+                return builder
+                    .Append(httpContext.GetResponseCachingState().CachedResponse.Guid.IdString)
```

Comment:
It's probably more efficient to use `string.Concat` here since there's only 3 parts.